src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -141,4 +170,130 @@
+            currentEra = [eraComponents era];
+            if (currentEra == era)
+            {
+                for (int day = 0; day < 31; day++)
+                {
+                    // subtract 1 day at a time until we get out of the specified Era
+                    startDateComponents.day = startDateComponents.day - 1;
+                    date = [japaneseCalendar dateFromComponents:startDateComponents];
+                    eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+                    currentEra = [eraComponents era];
+                    if (currentEra != era)
+                    {
+                        // add back 1 day to get back into the specified Era
+                        startDateComponents.day = startDateComponents.day + 1;
+                        date = [japaneseCalendar dateFromComponents:startDateComponents];
+                        NSCalendar *gregorianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+                        NSDateComponents *components = [gregorianCalendar components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:date];
+                        *startYear = [components year];
+                        *startMonth = [components month];
+                        *startDay = [components day];
+                        return 1;
+                    }
+                }
+            }
+            // add 1 month at a time until we get into the specified Era
+            startDateComponents.month = startDateComponents.month + 1;
+            date = [japaneseCalendar dateFromComponents:startDateComponents];
+            eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+            currentEra = [eraComponents era];
+        }
+
+        return 0;
+    }
+}
+
+/*
+Function:
+GetCalendarsNative
+
+Returns the list of CalendarIds that are available for the specified locale.
+*/
+int32_t GlobalizationNative_GetCalendarsNative(const char* localeName, CalendarId* calendars, int32_t calendarsCapacity)
+{
+    @autoreleasepool
+    {
+        NSArray *calendarIdentifiers = @[
+            NSCalendarIdentifierGregorian,
+            NSCalendarIdentifierBuddhist,
+            NSCalendarIdentifierChinese,
+            NSCalendarIdentifierCoptic,
+            NSCalendarIdentifierEthiopicAmeteMihret,
+            NSCalendarIdentifierEthiopicAmeteAlem,
+            NSCalendarIdentifierHebrew,
+            NSCalendarIdentifierISO8601,
+            NSCalendarIdentifierIndian,
+            NSCalendarIdentifierIslamicCivil,
+            NSCalendarIdentifierIslamicTabular,
+            NSCalendarIdentifierIslamicUmmAlQura,
+            NSCalendarIdentifierIslamic,
+            NSCalendarIdentifierJapanese,
+            NSCalendarIdentifierPersian,
+            NSCalendarIdentifierRepublicOfChina,
```

Comment:
I'm realizing that half of these will lead to an `UNINITIALIZED_VALUE` as a `CalendarId` because there is missing compatibility between the supported calendar identifiers on the native side and what is supported on the [managed CalendarId side](https://github.com/dotnet/runtime/blob/f37e653ee77490f2ce0ad488db213827fb50b7ff/src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormatInfoScanner.cs#L39-L68).

No `CalendarId` or `src/libraries/System.Private.CoreLib/src/System/Globalization/*Calendar.cs` parallel
```
NSCalendarIdentifierChinese,
NSCalendarIdentifierCoptic,
NSCalendarIdentifierEthiopicAmeteMihret,
NSCalendarIdentifierEthiopicAmeteAlem,
NSCalendarIdentifierISO8601,
NSCalendarIdentifierIndian,
NSCalendarIdentifierIslamicCivil,
NSCalendarIdentifierIslamicTabular,
```

I'm wondering if there should be more calendars added to the managed side to match these identifiers. What are the differences between these identifiers and the `NSCalendarIdentifierGregorian` behaviorwise on the native side?

Reply:
If we discover significant differences between those `NSCalendarIdentifier`s without a managed parallel and `NSCalendarIdentifierGregorian`, then we can figure out if it makes sense to add more to the managed side `CalendarId` enum and `src/libraries/System.Private.CoreLib/src/System/Globalization/*Calendar.cs`.

In the meantime, I think it would make more sense to just remove the 8 that don't have any matches because they dont influence the resulting `calendars[]` or `calendarCount`.

Reply:
Thanks, I will remove not used calendar types. Opened https://github.com/dotnet/runtime/issues/93199 for further investigation to see how these calendars are different from `NSCalendarIdentifierGregorian` and if there is a need to implement them.